examples/keccak.rs

```diff
@@ -0,0 +1,434 @@
+                    &mut cs.namespace(|| format!("alloc bit {i}")),
+                    if e.hash().get_value().unwrap_or(F::ZERO) == F::ZERO {
+                        Some(false)
+                    } else {
+                        Some(true)
+                    },
+                )
+                .unwrap()
+            })
+            .collect::<Vec<_>>();
+
+        // Padding to 256 bits
+        elts_alloc_bits.extend_from_slice(
+            &(0..256 - self.n * 8)
+                .map(|i| {
+                    AllocatedBit::alloc(&mut cs.namespace(|| format!("alloc bit {i}")), Some(false))
+                        .unwrap()
+                })
+                .collect::<Vec<_>>(),
+        );
+
+        Ok(vec![CircomInput::new(
+            "in".parse().unwrap(),
+            elts_alloc_bits
+                .iter()
+                .map(|e| {
+                    if e.get_value().unwrap_or(true) {
+                        F::ONE
+                    } else {
+                        F::ZERO
+                    }
+                })
+                .collect::<_>(),
+        )])
+    }
+
+    fn evaluate_simple(&self, s: &Store<F>, args: &[Ptr]) -> Ptr {
+        let (bits, _) = s.fetch_list(&args[0]).unwrap();
+
+        let z_bits = bits.iter().map(|ptr| s.hash_ptr(ptr)).collect::<Vec<_>>();
+
+        let bytes_to_hash = bits_to_bytes(
+            &z_bits
+                .iter()
+                .map(|ptr| ptr.value() != &F::ZERO)
+                .collect::<Vec<_>>(),
+        );
+
+        // Calculate the Keccak-256 hash
+        let mut hasher = Keccak::v256();
+        hasher.update(&bytes_to_hash);
+        let mut output_bytes = [0u8; 32];
+        hasher.finalize(&mut output_bytes);
+        // Convert the output bytes into a Vec<u8> representing bits
+        let output_bits_ptr = bytes_to_bits(&output_bytes)
+            .iter()
+            .map(|b| if *b { s.num(F::ONE) } else { s.num(F::ZERO) })
+            .collect::<Vec<Ptr>>();
+
+        s.list(output_bits_ptr)
+    }
+
+    fn arity(&self) -> usize {
+        1
+    }
+}
+
+fn main() {
+    // Get command line input
+    let args = std::env::args().collect::<Vec<_>>();
+
+    // Initialize store, responsible for handling variables in the lurk context
+    let store: &Store<pallas::Scalar> = &Store::default();
+
+    // Define the symbol that will call upon our Coprocessor
+    let str_to_le_bits_sym = user_sym("str_to_le_bits");
+    let keccak_sym = user_sym("keccak_hash");
+    let program = lurk_program(store, &args[1]);
+
+    // Create the Lang. ie the list of corprocessor that will be accessible in our program
+    let mut lang = Lang::<pallas::Scalar, KeccakExampleCoproc<pallas::Scalar>>::new();
+    lang.add_coprocessor(str_to_le_bits_sym, StrToBits::new(args[1].len()));
+    lang.add_coprocessor(
+        keccak_sym,
+        CircomCoprocessor::new(CircomKeccak::new(args[1].len())),
+    );
+    let lang_rc = Arc::new(lang.clone());
```

Comment:
Nit: this clone is not necessary as you can use `&lang_rc` where a `Lang` reference is needed

Reply:
Yes, I also switched to `Instance::new_supernova` for the primary instance